pkg/operator/staticpod/installerpod/cmd.go

```diff
@@ -136,144 +152,188 @@ func (o *InstallOptions) prefixFor(name string) string {
 				return err
 			}
 		}
 	}
-	for _, configmap := range configmaps {
+	for _, configmap := range configs {
 		contentDir := path.Join(resourceDir, "configmaps", o.prefixFor(configmap.Name))
-		glog.Infof("creating dir %q", contentDir)
+		glog.Infof("Creating directory %q ...", contentDir)
 		if err := os.MkdirAll(contentDir, 0755); err != nil {
 			return err
 		}
 		for filename, content := range configmap.Data {
-			glog.Infof("writing configmap file %q", path.Join(contentDir, filename))
+			glog.Infof("Writing config file %q ...", path.Join(contentDir, filename))
 			if err := ioutil.WriteFile(path.Join(contentDir, filename), []byte(content), 0644); err != nil {
 				return err
 			}
 		}
 	}
+
 	podFileName := o.PodConfigMapNamePrefix + ".yaml"
+	glog.Infof("Writing pod manifest %q ...", path.Join(resourceDir, podFileName))
 	if err := ioutil.WriteFile(path.Join(resourceDir, podFileName), []byte(podContent), 0644); err != nil {
 		return err
 	}
 
 	// copy static pod
-	glog.Infof("creating dir %q", o.PodManifestDir)
+	glog.Infof("Creating directory for static pod manifest %q ...", o.PodManifestDir)
 	if err := os.MkdirAll(o.PodManifestDir, 0755); err != nil {
 		return err
 	}
 
 	for _, fn := range o.PodMutationFns {
-		glog.V(2).Infof("customizing static pod")
+		glog.V(2).Infof("Customizing static pod ...")
 		pod := resourceread.ReadPodV1OrDie([]byte(podContent))
 		if err := fn(pod); err != nil {
 			return err
 		}
 		podContent = resourceread.WritePodV1OrDie(pod)
 	}
-	glog.Infof("writing static pod %q", path.Join(o.PodManifestDir, podFileName))
-	glog.Infof("writing static pod\n%s", podContent)
+
+	o.NodeName, err = getPodNodeName([]byte(podContent))
+	if err != nil {
+		return fmt.Errorf("failed to get pod node name: %v", err)
+	}
+
+	glog.Infof("Writing static pod manifest %q ...\n%s", path.Join(o.PodManifestDir, podFileName), podContent)
 	if err := ioutil.WriteFile(path.Join(o.PodManifestDir, podFileName), []byte(podContent), 0644); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (o *InstallOptions) Run() error {
-	// ~2 min total waiting
-	backoff := utilwait.Backoff{
-		Duration: time.Second,
-		Factor:   1.5,
-		Steps:    11,
+func getPodNodeName(podBytes []byte) (string, error) {
+	podJSONBytes, err := yaml.YAMLToJSON(podBytes)
+	if err != nil {
+		return "", err
 	}
-	attempts := 0
-	err := utilwait.ExponentialBackoff(backoff, func() (bool, error) {
-		attempts += 1
-		if copyErr := o.copyContent(); copyErr != nil {
-			fmt.Fprintf(os.Stderr, "#%d: failed to copy content: %v", attempts, copyErr)
-			return false, nil
-		}
-		return true, nil
-	})
+	unstructuredPodObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, podJSONBytes)
 	if err != nil {
-		return fmt.Errorf("error: %v", err)
+		return "", err
 	}
+	nodeName, _, err := unstructured.NestedString(unstructuredPodObj.(*unstructured.Unstructured).Object, "spec", "nodeName")
```

Comment:
nit: the 2nd return value tells you whether the fields existed. Might be worth to check for `len(nodeName) != 0`.

Reply:
i think we can return "<unknown>" in that case?

Reply:
This won't happen anyway.